src/mlpack/methods/ann/activation_functions/softplus_function.hpp

```diff
@@ -51,9 +51,21 @@ class SoftplusFunction
    */
   static double Fn(const double x)
   {
+    const double val = std::log(1 + std::exp(x));
     if (x < DBL_MAX)
-      return x > -DBL_MAX ? std::log(1 + std::exp(x)) : 0;
-    return 1.0;
+    {
+      if (x > -DBL_MAX)
+      {
+        if (std::isfinite(val))
+          return val;
+        else
+          return x;
```

Comment:
Please make use of additional braces `{}` with if-else statements.

Reply:
Actually I'm note sure.. Its written `To conserve lines, refrain from using braces if the if, while, or for statement applies to a single statement.` [here](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#brace-placement).. Should I add braces?

Reply:
If you place braces right next to the `if-else` endings, you'll only require one extra line. And that's fine, if it improves readability. Compilers are indifferent to the number of lines used.